app/controllers/passengers_controller.rb

```diff
@@ -0,0 +1,75 @@
+class PassengersController < ApplicationController
+
+  def index
+    @passengers = Passenger.all
+  end
+
+  def show
+    @passenger = Passenger.find_by(id: params[:id])
+    if @passenger.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def new
+    @passenger = Passenger.new
+    if params[:passenger_id].nil?
+      @trip = Trip.new
+    else
+      passenger = passenger.find_by(id: params[:id])
+      passenger.trips.new
+      return
+    end
+  end
+
+  def create
+    @passenger = Passenger.new(passenger_params)
+    if @passenger.save
+      redirect_to passenger_path(@passenger.id)
+      return
+    else
+      render :new
+      return
+    end
+  end
+
+  def edit
+    @passenger = Passenger.find_by(id: params[:id])
+    if @passenger.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def update
+    @passenger = Passenger.find_by(id: params[:id])
+    if @passenger.nil?
+      head :not_found
+      return
+    elsif @passenger.update(passenger_params)
+      redirect_to passenger_path
+      return
+      render :edit, status: :bad_request
+      return
+    end
+  end
+
+  def destroy
+    @passenger = Passenger.find_by(id: params[:id])
+    if @passenger.nil?
+      head :not_found
+      return
+    else
+      @passenger.destroy
+      redirect_to passengers_path
+      return
+    end
+  end
```

Comment:
There's an issue here if you destroy a passenger when it has a trip, the trip still exists, but the passenger doesn't.

There's a couple of ways to handle this:

1.  Delete all the trips belonging to the passenger.
1.  Make the passenger "inactive" somehow, still in the DB but doesn't show up in the lists.  Maybe just have a field active/inactive in the Passenger model.

BTW this also applies to driver.